libkbfs/random_backoff.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2016 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"math/rand"
+	"sync"
+	"time"
+)
+
+var randomBackoffRand *rand.Rand
+
+func init() {
+	randomBackoffRand = rand.New(rand.NewSource(time.Now().Unix()))
+}
+
+type fireOnce struct {
+	ch   chan struct{}
+	once sync.Once
+}
+
+func newFireOnce() *fireOnce {
+	return &fireOnce{ch: make(chan struct{})}
+}
+
+func (o *fireOnce) fire() {
+	if o == nil {
+		return
+	}
+	o.once.Do(func() { close(o.ch) })
+}
+
+func (o *fireOnce) wait() {
+	if o == nil {
+		return
+	}
+	<-o.ch
+}
+
+// CancelableRandomBackoff can be used to wait on a random backoff timer. A
```

Comment:
Also, it's not a big deal, but in `delayed_cancellation.go`, all of the forms of "cancel" are spelled with two Ls in the middle, and this one has only one L.  Would be nice to be consistent.  (I think both spellings are technically correct, so it doesn't really matter which.)